make/protogen.mk

```
@@ -37,15 +37,16 @@ endif
 PROTOC_VERSION := 21.4
 UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S),Linux)
-PROTOC_ARCH = linux
+PROTOC_OS = linux
 endif
 ifeq ($(UNAME_S),Darwin)
-PROTOC_ARCH = osx
+PROTOC_OS = osx
 endif
+PROTOC_ARCH=$(shell case $$(uname -m) in (aarch64) echo -n armv6 ;; (*) echo -n $$(uname -m) ;; esac)
 
 PROTO_PRIVATE_DIR := $(BASE_PATH)/.proto
 
-PROTOC_DIR := $(PROTO_PRIVATE_DIR)/protoc-$(PROTOC_ARCH)-$(PROTOC_VERSION)
+PROTOC_DIR := $(PROTO_PRIVATE_DIR)/protoc-$(PROTOC_OS)-$(PROTOC_VERSION)
```

I suggest including both OS and Arch to the directory name:

```suggestion
PROTOC_DIR := $(PROTO_PRIVATE_DIR)/protoc-$(PROTOC_OS)-$(PROTOC_ARCH)-$(PROTOC_VERSION)
```